app/src/main/java/com/seafile/seadroid2/transfer/UploadTask.java

```diff
@@ -80,14 +87,26 @@ public boolean isCancelled() {
                 }
             };
             if (isUpdate) {
-                dataManager.updateFile(repoName, repoID, dir, path, monitor, isCopyToLocal);
+                if (byBlock) {
+                    dataManager.updateByBlocks(repoName, repoID, dir, path, monitor, isCopyToLocal, version);
+                } else {
+                    dataManager.updateFile(repoName, repoID, dir, path, monitor, isCopyToLocal);
+                }
             } else {
-                Log.d(DEBUG_TAG, "Upload path: " + path);
-                dataManager.uploadFile(repoName, repoID, dir, path, monitor, isCopyToLocal);
+                if (byBlock) {
+                    dataManager.uploadByBlocks(repoName, repoID, dir, path, monitor, isCopyToLocal, version);
+                } else {
+                    dataManager.uploadFile(repoName, repoID, dir, path, monitor, isCopyToLocal);
+                }
```

Comment:
AFAIK, I can\`t just use "upload/download-by-block" for all, because file transfer and block transfer don\`t have the same implementations. But I removed the duplicate  "update" code, and only kept the "upload" code. Maybe it could be refactor much deeper, I will keep thinking on that.